app/desktop/studio_server/tool_servers_api.py

```diff
@@ -21,6 +23,10 @@ class ExternalToolServerCreationRequest(BaseModel):
     description: str | None = None
 
 
+class ExternalToolServerApiDescription(ExternalToolServer):
+    available_tools: list[Tool]
```

Comment:
Don't use the mcp package object to define our API. We don't want a 3rd party controlling our API. Plus it needs to be a Pydantic model so I'm surprised it works at all. And we're trying to allow non-mcp externalToolServers in the future.

Make a new `class ToolApiDecription(BaseClass)` with the fields you want. Can add a class method that takes a MCP server and returns a ToolApiDecription.

```
  @classmethod
  def tool_from_mcp_server(cls, mcpTool Tool):
    return ToolApiDecription(
       name=mcpTool.name,
       description=mcpTool.description,
       ...
  )
```

Then call it like
```
ToolApiDecription.tool_from_mcp_server(mcpTool)
```